From 20f022cd1ee2d51b8c28f039b82341cf6b008dc8 Mon Sep 17 00:00:00 2001
From: Jon West <electrikjesus@gmail.com>
Date: Tue, 11 Jun 2024 10:39:06 -0400
Subject: [PATCH 08/14] InputDevice: Allow to ignore external display input

Uses ro.boot.force.ignore_external_input as the property switch
---
 services/inputflinger/reader/InputDevice.cpp | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/services/inputflinger/reader/InputDevice.cpp b/services/inputflinger/reader/InputDevice.cpp
index 957604faf6..3c6914c00c 100644
--- a/services/inputflinger/reader/InputDevice.cpp
+++ b/services/inputflinger/reader/InputDevice.cpp
@@ -254,7 +254,11 @@ std::list<NotifyArgs> InputDevice::configureInternal(nsecs_t when,
 
     using Change = InputReaderConfiguration::Change;
 
-    if (!changes.any() || !isIgnored()) {
+    // Allow blocking external device input
+    char mIgnoreExternalInput[PROPERTY_VALUE_MAX] = {0};
+    property_get("ro.boot.force.ignore_external_input", mIgnoreExternalInput, "0");
+
+    if ((!changes.any() || !isIgnored()) && ((strcmp(mIgnoreExternalInput, "1") == 0) && mIsExternal)) {
         // Full configuration should happen the first time configure is called
         // and when the device type is changed. Changing a device type can
         // affect various other parameters so should result in a
-- 
2.43.0

